test/node-rpc-test.js

```diff
@@ -5,59 +5,512 @@
+
+      await assert.rejects(async () => {
+        await nclient.execute('gettxoutproof', [[txid], blockhash]);
+      }, {
+        type: 'RPCError',
+        message: 'Block does not contain all txids.',
+        code: -25 >>> 0
+      });
+    });
+
+    it('should verify in spv mode and pruned mode', async () => {
+      const txid = txids[2][0];
+      const proof = await nclient.execute('gettxoutproof', [[txid]]);
+
+      for (const client of [spvnclient, prunednclient]) {
+        const verify = await client.execute('verifytxoutproof', [proof]);
+
+        assert(verify);
+        assert.strict(verify.length, 1);
+        assert.strict(verify[0], txid);
+      }
+    });
+
+    it('should create proof for spent coin with txindex', async () => {
+      const txid = txids[0][0];
+      const res = await indexednclient.execute('gettxoutproof', [[txid]]);
+      const verify = await nclient.execute('verifytxoutproof', [res]);
+
+      assert(verify);
+      assert.strictEqual(verify.length, 1);
+      assert.strictEqual(verify[0], txid);
+    });
+
+    it('should create proof for one tx w/o blockhash', async () => {
+      const check = [
+        txids[2][0],
+        txids[2][1],
+        txids[3][0],
+        txids[3][1]
+      ];
+
+      for (const txid of check) {
+        const proof = await nclient.execute('gettxoutproof', [[txid]]);
+        const verify = await nclient.execute('verifytxoutproof', [proof]);
+
+        assert(verify);
+        assert.strictEqual(verify.length, 1);
+        assert.strictEqual(verify[0], txid);
+      }
+    });
+
+    it('should create proof one tx w/o blockhash (merkle)', async () => {
+      const txid = txids[2][2];
+
+      const proof = await nclient.execute('gettxoutproof', [[txid]]);
+      const verify = await nclient.execute('verifytxoutproof', [proof]);
+
+      assert(verify);
+      assert.strictEqual(verify.length, 1);
+      assert.strictEqual(verify[0], txid);
+
+      const mblock = MerkleBlock.fromRaw(proof, 'hex');
+      const mblockJSON = mblock.toJSON();
+      const tree = mblock.getTree();
+
+      assert.strictEqual(mblockJSON.hash, blockhashes[2]);
+      assert.strictEqual(mblockJSON.prevBlock, blockhashes[1]);
+      assert.strictEqual(mblockJSON.totalTX, txids[2].length + 1);
+      assert.bufferEqual(mblock.merkleRoot, tree.root);
+      assert.strictEqual(tree.matches.length, 1);
+    });
+
+    it('should create proof of multiple txs in a block', async () => {
+      const blockhash = blockhashes[3];
+      const list = [
+        txids[3][0],
+        txids[3][5]
+      ];
+
+      const proof = await nclient.execute('gettxoutproof', [list, blockhash]);
+      const verify = await nclient.execute('verifytxoutproof', [proof]);
+
+      assert(verify);
+      assert.strictEqual(verify.length, 2);
+      assert.ok(verify.includes(list[0]));
+      assert.ok(verify.includes(list[1]));
+    });
   });
 });
+
+function getNodePorts(ports) {
```

Comment:
Why not format this like `getClientOpts` and put at the top with the other helpers?